File: JavaScript/estoque/estoque.js
import Produto from "./entities/Produto.js";
import * as readline from 'node:readline';

async function perguntar(pergunta) {

    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    return new Promise(resolve => {
        rl.question(pergunta, (resposta) => {
            rl.close();
            resolve(resposta);
        })
    });
}

const nome = await perguntar("Nome: ");
const preco = await perguntar("Preço: ");
const produto = new Produto(nome, preco);

produto.adicionarAoEstoque(await perguntar("Quantidade em estoque: "));
console.log("Dados do produto: " + produto.toString());

produto.adicionarAoEstoque(await perguntar("Entre com a quantidade a ser adicionada ao estoque: "));
console.log("Produto adicionado ao estoque: " + produto.toString());

produto.removerDoEstoque(await perguntar("Entre com a quantidade a ser removida do estoque: "));
console.log("Produto removido do estoque: " + produto.toString());

File: Java/estoque/src/application/estoque.java
package application;

import java.util.Scanner;

import entities.Produto;

public class estoque {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		System.out.println("Entre com os dados do produto: ");
		
		System.out.println("Nome: ");
		String nome = sc.nextLine();
		
		System.out.println("Preço: ");
		double preco = sc.nextDouble();
		
		Produto produto = new Produto(nome, preco);
		
		System.out.println("Quantidade em estoque: ");
		produto.adicionarAoEstoque(sc.nextInt());
		
		System.out.println("Dados do produto: " + produto);
		
		System.out.println("Entre com a quantidade a ser adicionada ao estoque: ");
		produto.adicionarAoEstoque(sc.nextInt());
		
		System.out.println("Produto adicionado ao estoque: " + produto);
		
		System.out.println("Entre com a quantidade a ser removida do estoque: ");
		produto.removerDoEstoque(sc.nextInt());
		
		System.out.println("Produto removido do estoque: " + produto);
		
		sc.close();
	}

}